test/connectivity/NetworkController/Program.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace NetworkController
+{
+    using System;
+    using System.Net.NetworkInformation;
+    using System.Runtime.Loader;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Docker.DotNet;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<Program>();
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            var cts = new CancellationTokenSource();
+            AssemblyLoadContext.Default.Unloading += ctx => cts.Cancel();
+            Console.CancelKeyPress += (sender, cpe) => cts.Cancel();
+            var tcs = new TaskCompletionSource<bool>();
+            cts.Token.Register(s => ((TaskCompletionSource<bool>)s).SetResult(true), tcs);
+
+            switch (Settings.Current.Mode)
+            {
+                case Mode.Offline:
+                    StartOfflineMode();
+                    break;
+                case Mode.Satellite:
+                    StartSatelliteMode();
+                    break;
+            }
+
+            await tcs.Task;
+            return 0;
+        }
+
+        static void StartSatelliteMode() => throw new NotImplementedException();
+
+        static void StartOfflineMode()
+        {
+            var delay = Settings.Current.StartAfter;
+            var networkInterfaceName = GetDockerInterfaceName();
+            networkInterfaceName.Match(
```

Comment:
Nit: Instead of returning Task from match, you can use the if(!option.HasValue){} else {option.ForEach()} pattern.